app/src/main/java/com/kickstarter/viewmodels/CampaignDetailsViewModel.kt

```diff
@@ -1,42 +1,79 @@
 package com.kickstarter.viewmodels
 
-import androidx.annotation.NonNull
+import android.util.Pair
 import com.kickstarter.libs.ActivityViewModel
 import com.kickstarter.libs.Environment
-import com.kickstarter.models.Project
+import com.kickstarter.libs.models.OptimizelyExperiment
+import com.kickstarter.libs.rx.transformers.Transformers.combineLatestPair
+import com.kickstarter.models.User
 import com.kickstarter.ui.IntentKey
 import com.kickstarter.ui.activities.CampaignDetailsActivity
+import com.kickstarter.ui.data.ProjectData
 import rx.Observable
 import rx.subjects.BehaviorSubject
+import rx.subjects.PublishSubject
 
 interface CampaignDetailsViewModel {
 
-    interface Inputs
+    interface Inputs {
+        /** Call when the user clicks the pledge action button. */
+        fun pledgeActionButtonClicked()
+    }
 
     interface Outputs {
+        /** Emits when we should return to the [com.kickstarter.ui.activities.ProjectActivity] with the rewards visible. */
+        fun goBackToProject(): Observable<Void>
+
+        /** Emits a boolean determining if the faux pledge container is visible. */
+        fun pledgeContainerIsVisible(): Observable<Boolean>
```

Comment:
Ooh, not isGone? 😄

Reply:
I think `isVisible` is less confusing but should I do `isGone` to be consistently confusing?

Reply:
Yeah, in our case we just match it to the property it's setting which is `isHidden`. Whatever works!